internal/bminventory/inventory.go

```diff
@@ -529,14 +553,17 @@ func (b *bareMetalInventory) validateRegisterClusterInternalPreDefaultValuesSet(
 	params.NewClusterParams.HighAvailabilityMode, params.NewClusterParams.ControlPlaneCount = common.GetDefaultHighAvailabilityAndMasterCountParams(
 		params.NewClusterParams.HighAvailabilityMode, params.NewClusterParams.ControlPlaneCount,
 	)
+	if params.NewClusterParams.OpenshiftVersion == nil {
+		return common.NewApiError(http.StatusBadRequest, errors.New("OpenshiftVersion is required"))
+	}
 	if err := b.validateHighAvailabilityWithControlPlaneCount(*params.NewClusterParams.HighAvailabilityMode, *params.NewClusterParams.ControlPlaneCount, *params.NewClusterParams.OpenshiftVersion); err != nil {
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
 	if err := validations.ValidateClusterCreateIPAddresses(b.IPv6Support, id, params.NewClusterParams); err != nil {
 		b.log.WithError(err).Error("Cannot register cluster. Failed VIP validations")
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
-	if err := validations.ValidateDualStackNetworks(params.NewClusterParams, false, false); err != nil {
+	if err := validations.ValidateDualStackNetworks(params.NewClusterParams, false, false, *params.NewClusterParams.OpenshiftVersion); err != nil {
```

Comment:
For possible code changes in the future, I would still use `swagger.StringValue`